# Tasks                          

## Task 1: Source the data set

# question -- Import the Iris data set from the sklearn.datasets module

In [ ]:



# firstly import the pandas function to read comma seperated values (csv), which will be in the data set
# rename to pd for short

import pandas as pd

# import the sklearn package which is a machine learning libraries for Python.good for practicing on datasets.
# rename to skl for short

import sklearn as skl

#load the iris dataset
#df is a pandas fuction that will display dataframe

df=pd.read_csv('https://gist.githubusercontent.com/curran/a08a1080b88344b0c8a7/raw/0e7a9b0a5d22642a06d3d5b9bcbad9890c8ee534/iris.csv')

# df to display
df


#Reference: pandas- "https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html"
#Reference: Ian McLoughlin Lecture : week 3 practical-pands and loading datasets

sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

# question:  Explain, in your own words, what the load_iris() function returns.



load_iris() is a panda function that enables you to load in data from Scikit-learn , and you can then chose what specific data you would like to be displayed like in the example below with "feature_names"

"load_iris() is a a quick way to load the popular iris dataset from Scikit-learn and it is used to practicing classification algorithms and exploratory data analysis."
Reference: chatgpt- prompt-explain load_iris(), "https://chatgpt.com/c/6813ac34-43f0-8003-b08c-170c373f4fb8"

Reference: Ian McLoughlin Lecture : week 3 practical-pands and loading datasets

In [ ]:
import pandas as pd
import sklearn as skl
from sklearn.datasets import load_iris

#how to display the dataset imported-
#rename function to iris

iris = load_iris()

#example of displaying information from set like feature names

print(iris.feature_names)    

['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


## Task 2: Explore the data structure.

##- Print and explain the shape of the data set, the first and last 5 rows of the data, the feature names, and the target classes.

In [ ]:
# to do this i will need to use other panda functions
# all which can be found on the pandas website- Reference: "https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html"
# also Chatgpt:prompt- to get the last five rows, "https://chatgpt.com/c/6813ac34-43f0-8003-b08c-170c373f4fb8"

# for the last 5 rows use : print to display, df for dataframe, and for the last 5 rows - .tail() function

print(df.tail())



     sepal_length  sepal_width  petal_length  petal_width    species
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica


explaining the data shown- in last 5 rows as shown above the function is displaying the last 5 rows of data from specimen 145 to 149 , aslo includes the column for values in length, width for sepal and length and width for petal , also the species column

same as above but for the feature name, also from the same sources- iris.feature_names function

In [27]:
iris.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

explaining the data shown- sepal length and width and petal length and width in centimetres are the feature names 

same as above but for the target classes, also from the same sources-iris.target_names function

In [28]:
iris.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

explaining the data shown- array is all the target names(classes) in the dataset which in this case are setosa,versicolor and virginica 

## Task 3: summarize the data

# For each feature in the dataset, calculate and display:

   - mean
   -minimum
   -maximum
   -standard deviation
   -median


## Task 4. Visualize Features

 Plot histograms for each feature using matplotlib.
 Add appropriate titles and axis labels